src/skmultiflow/evaluation/evaluate_prequential_delayed.py

```diff
@@ -0,0 +1,514 @@
+                    self.model[i].partial_fit(X=X, y=y)
+                    self.running_time_measurements[i].compute_training_time_end()
+                self.running_time_measurements[i].update_time_measurements(self.pretrain_size)
+            self.global_sample_count += self.pretrain_size
+            self.first_run = False
+            # initialize time_manager with last timestamp avaiable
+            self.time_manager = TimeManager(arrival_time.to_list()[-1])
+
+        self.update_count = 0
+        print('Evaluating...')
+        while ((self.global_sample_count < self.actual_max_samples) & (
+                self._end_time - self._start_time < self.max_time)
+               & (self.stream.has_more_samples())):
+            try:
+
+                # get current batch
+                current_batch = self.stream.next_sample(self.batch_size)
+
+                # TODO: improve this solution for more optional parameters than just weight. Also, include weight in
+                #  the delayed_queue check if batch contains weight (change here if we include more informations in
+                #  TemporalDataStream)
+                if len(current_batch) > 4:
+                    X, arrival_time, available_time, y_real, weight = current_batch
+                else:
+                    X, arrival_time, available_time, y_real = current_batch
+
+                # update current timestamp
+                self.time_manager.update_timestamp(arrival_time.to_list()[-1])
+
+                # get delayed samples to update model before predicting a new batch
+                X_delayed, y_real_delayed, y_pred_delayed = self.time_manager.get_available_samples()
+
+                self._update_metrics_delayed(y_real_delayed, y_pred_delayed)
+
+                # before getting new samples, update classifiers with samples that are already available
+                self._update_classifiers(X_delayed, y_real_delayed)
+
+                # predict samples and get predictions
+                y_pred = self._predict_samples(X)
+
+                # add current samples to delayed queue
+                self.time_manager.update_queue(X, arrival_time, available_time, y_real, y_pred)
+
+                self._end_time = timer()
+            except BaseException as exc:
+                print(exc)
+                if exc is KeyboardInterrupt:
+                    self._update_metrics()
+                break
+
+        # evaluate remaining samples in the delayed_queue
+        # iterate over delay_queue while it has samples according to batch_size
+        while self.time_manager.has_more_samples():
+            # get current samples to process
+            X_delayed, y_real_delayed, y_pred_delayed = self.time_manager.next_sample(self.batch_size)
+            # update metrics
+            self._update_metrics_delayed(y_real_delayed, y_pred_delayed)
+            # update classifier with these samples
+            # TODO: does it make sense?
```

Comment:
Nit: Would you please clarify the Todo note

Reply:
Fixed it in commit [6c3383071d6ac6ec8d36d2eb67203bd44f44f27c](https://github.com/scikit-multiflow/scikit-multiflow/commit/6c3383071d6ac6ec8d36d2eb67203bd44f44f27c). This TODO was only to remember me about updating classifiers after checking the remaining samples in the delayed queue (if it makes sense). In a meeting, we agreed that it makes sense, given that evaluation returns the models.